### PR TITLE
Remove hostname and port from HttpClientDecorator

### PR DESCRIPTION
**Pull request by aarya123**
Was talking to @tylerbenson and saw that out profiles were doing a lot of parsing of URIs. I've found the culprit to be in the `hostname(REQUEST request)` and `port(REQUEST request)` methods. Since this data is already computed via the initial URI parse to build the url, I see no need to delegate the extra calls for tags.

### CODE REVIEW

**Comment by aarya123** [comment]
I don't know much of groovy, so I'm just gonna make random changes until it works. lmk if I'm on the wrong track

**Comment by aarya123** [comment]
First try! ;)

**Comment by aarya123** [comment]
woe is me...

File: dd-java-agent/instrumentation/aws-java-sdk-1.11.0/src/test/groovy/AWSClientTest.groovy [hunk 1]
```diff
@@ -1,16 +1,5 @@
...
-import com.amazonaws.SdkClientException
-import com.amazonaws.auth.AWSCredentialsProviderChain
-import com.amazonaws.auth.AWSStaticCredentialsProvider
-import com.amazonaws.auth.AnonymousAWSCredentials
-import com.amazonaws.auth.BasicAWSCredentials
-import com.amazonaws.auth.EnvironmentVariableCredentialsProvider
-import com.amazonaws.auth.InstanceProfileCredentialsProvider
-import com.amazonaws.auth.SystemPropertiesCredentialsProvider
+import com.amazonaws.*
+import com.amazonaws.auth.*
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/aws-java-sdk-1.11.0/src/test/groovy/AWSClientTest.groovy]: We avoid having wildcards... you can increase your threshold in your ide.

File: dd-java-agent/agent-bootstrap/src/test/groovy/datadog/trace/bootstrap/instrumentation/decorator/HttpClientDecoratorTest.groovy [hunk 2]
```diff
@@ -28,10 +28,10 @@ class HttpClientDecoratorTest extends ClientDecoratorTest {
...
       1 * span.setTag(Tags.HTTP_METHOD.key, req.method)
       1 * span.setTag(Tags.HTTP_URL.key, "$req.url")
-      1 * span.setTag(Tags.PEER_HOSTNAME.key, req.host)
-      1 * span.setTag(Tags.PEER_PORT.key, req.port)
+      1 * span.setTag(Tags.PEER_HOSTNAME.key, req.url.getHost())
+      1 * span.setTag(Tags.PEER_PORT.key, req.url.getPort())
       if (renameService) {
-        1 * span.setTag(DDTags.SERVICE_NAME, req.host)
+        1 * span.setTag(DDTags.SERVICE_NAME, req.url.getHost())
       }
```
Comment (tylerbenson) [inline dd-java-agent/agent-bootstrap/src/test/groovy/datadog/trace/bootstrap/instrumentation/decorator/HttpClientDecoratorTest.groovy]: Groovy will allow the following:
```groovy
      1 * span.setTag(Tags.PEER_HOSTNAME.key, req.url.host)
      1 * span.setTag(Tags.PEER_PORT.key, req.url.port)
      if (renameService) {
        1 * span.setTag(DDTags.SERVICE_NAME, req.url.host)
      }
```

Standard getters (`get*()` methods) can be accessed like a field.

**Comment by tylerbenson** [comment]
The java 7 failure is unrelated.

**Pull request merged**